code/noRAM/app/src/main/java/com/example/noram/model/Photo.java

```diff
@@ -14,6 +28,10 @@
 public class Photo {
```

Comment:
want to make sure I understand the purpose of these changes here; you've added observer pattern to the photo class so ImageViews can listen be connected to a photo so that when it is updated they are updated.

This doesn't seem bad to me, but what specifically is the use for this? If an admin deletes a profile photo, we'd need to communicate that through the database to other devices, so this wouldn't exactly be able to update that. Is this intended for that part of the app, or just for the admin section?

Reply:
I made updateWithBitmap() because of the list (grid) of ImageViews in the admin's image section. 
When a view gets created in the list, it needs to be immediately displayed. However, queries to the database are asynchronous. Thus, while waiting for images to get fetched, a placeholder image is displayed until the database's image is loaded.

All of the changes I made were to implement the photos in the Admin section: if the changes are too heavy or unrelated however, I can also make an AdminPhoto subclass and move the implementation.

Reply:
Makes sense, I don't have any issue with whatever implementation you found to be best. I do think it might be good to rename the class/make it a bit clearer though - just using "Photo" as the class for this to me suggests that it would be used for photos across the app, when in reality this is specifically for the admin photos. It might be a good idea to rename the class (I don't think anything else is using the Photo class anyways?)